app/src/lib/merge.ts

```diff
@@ -1,5 +1,5 @@
 /** Create a copy of an object by merging it with a subset of its properties. */
-export function merge<T, K extends keyof T>(obj: T, subset: Pick<T, K>): T {
+export function merge<T extends {}, K extends keyof T>(obj: T | null | undefined, subset: Pick<T, K>): T {
```

Comment:
This fixes some typing issues with both the implementation and callers when `obj` is possibly null or undefined